internal/proxy/impl.go

```diff
@@ -7071,9 +7069,90 @@ func (node *Proxy) RunAnalyzer(ctx context.Context, req *milvuspb.RunAnalyzerReq
 			results[i].Tokens = append(results[i].Tokens, token)
 		}
 	}
+	return results, nil
+}
 
-	return &milvuspb.RunAnalyzerResponse{
-		Status:  merr.Status(nil),
-		Results: results,
-	}, nil
+func (node *Proxy) RunAnalyzer(ctx context.Context, req *milvuspb.RunAnalyzerRequest) (*milvuspb.RunAnalyzerResponse, error) {
+	ctx, sp := otel.Tracer(typeutil.ProxyRole).Start(ctx, "Proxy-RunAnalyzer")
+	defer sp.End()
+
+	if err := merr.CheckHealthy(node.GetStateCode()); err != nil {
+		return &milvuspb.RunAnalyzerResponse{
+			Status: merr.Status(err),
+		}, nil
+	}
+
+	if req.GetCollectionName() == "" {
+		results, err := node.runAnalyzer(req)
+		if err != nil {
+			return &milvuspb.RunAnalyzerResponse{
+				Status: merr.Status(err),
+			}, nil
+		}
+
+		return &milvuspb.RunAnalyzerResponse{
+			Status:  merr.Status(nil),
+			Results: results,
+		}, nil
+	}
+
+	if err := validateRunAnalyzer(req); err != nil {
```

Comment:
how about call validateRunAnalyzer right after entering this method, and add more checks in validateRunAnalyzer?

Reply:
can also check size of analyzer_names here